### ISSUE TITLE
Support for ZRLE encoding in VNC

### ISSUE THREAD

**Issue by AntonAndreevichMoroz** [Ylianst/MeshCentral#2603]
There is such a product WtWare (OS for thin clients). They have a built-in VNC server. I use the new Local Devices feature and noVNC to connect. But I get an error on the current client side. Having tried the same thing in Guacamole, the connection was successful for me. I started looking for information by mistake and sorting out.
Found a similar problem with noVNC (https://wtware.ru/forum/viewtopic.php?t=21514 unfortunately only in Russian). It indicates the need to have support for ZRLE encoding in VNC.
Then I found discussions about noVNC and this type of encoding. Which brought me to github noVNC https://github.com/novnc/noVNC/issues/545 and see that this issue has been resolved. I tried to compare what might be the differences between the original noVNC and what is built into MeshCentral. Found missing noVNC/core/deflator.js file. Unfortunately my knowledge is not enough to try to implement it.

I will give the connection logs through MeshCentral
```
10-19-31-932| [            gm] [  820.963493] [VNC] Connection from 192.168.200.8.
10-19-31-979| [            gm] [  821.005642] [VNC] Client VNC version 8 (RFB 003.008).
10-19-37-948| [            gm] [  826.977682] [VNC] Password ok.
10-19-38-010| [            gm] [  827.040134] [VNC] bits-per-pixel 32, depth 24, big-endian false, true-colour true.
10-19-38-057| [            gm] [  827.040244] [VNC] red-max 0x00ff, green-max 0x00ff, blue-max 0x00ff.
10-19-38-072| [            gm] [  827.040319] [VNC] red-shift 16, green-shift 8, blue-shift 0.
10-19-38-072| [            gm] [  827.040385] [VNC] 32-bit server, 32-bit client.
10-19-38-072| [            gm] [  827.069987] [VNC] 16 encodings: 1 7 -260 5 2 0 -26 -254 -223 -224 -258 -308 -309 -312 -313 -239.
```

I will give the connection logs through Guacamole
```
10-20-13-036| [            gm] [  862.069148] [VNC] Connection from 10.10.1.201.
10-20-13-083| [            gm] [  862.075333] [VNC] Client VNC version 8 (RFB 003.008).
10-20-13-083| [            gm] [  862.087309] [VNC] Password ok.
10-20-13-083| [            gm] [  862.097833] [VNC] bits-per-pixel 32, depth 24, big-endian false, true-colour true.
10-20-13-099| [            gm] [  862.097885] [VNC] red-max 0x00ff, green-max 0x00ff, blue-max 0x00ff.
10-20-13-099| [            gm] [  862.097924] [VNC] red-shift 16, green-shift 8, blue-shift 0.
10-20-13-099| [            gm] [  862.097963] [VNC] 32-bit server, 32-bit client.
10-20-13-114| [            gm] [  862.098625] [VNC] 19 encodings: 16 9 ffff0009 1 5 6 4 2 0 -253 -240 -239 -232 fffe0000 -223 fffe0001 fffe0002 fffe0003 -309.
```

Here we see differences only in encoding options. @Ylianst Could you check this issue?

**Comment by nzalev** [comment]
Guacamole specifically supports ZRLE. I might be wrong here, but it doesn't look like noVNC does. I found the same issue you linked at noVNC, but it's not currently resolved. It's been labeled as a Future Feature. A PR was submitted to noVNC to add support for ZRLE, (https://github.com/novnc/noVNC/pull/1283) but it was closed and not merged.

I took a look at the current list of encodings at noVNC, and ZRLE is still not in there.

**Comment by Ylianst** [comment]
I think you right, it's still no there. That is weird since I did the full implementation of the RFB protocol for Intel AMT and added ZRLE support in my viewer. Hard to believe noVNC still does not have it.

**Comment by AntonAndreevichMoroz** [comment]
@Ylianst Judging from this thread https://github.com/novnc/noVNC/pull/1589 discussions there are already changes on this issue.  And it seems that in recent updates, ZRLE support has been added.  Could you please update the noVNC module to add support?

**Comment by Ylianst** [comment]
Good catch! Started working on updating noVNC and indeed, it seems to have ZRLE support. Hopefully I will have this ready by tomorrow to be pushing into GitHub.

**Comment by Ylianst** [comment]
Just updated MeshCentral with noVNC 1.3.0. This will be in MeshCentral v0.9.55. Once published, let me know if it works.

**Comment by Ylianst** [comment]
Got the graphics ready for the release.

![MC2-noVNC130](https://user-images.githubusercontent.com/1319013/143939488-63d8138e-0448-4a08-b58c-946ec6708420.png)

**Comment by Ylianst** [comment]
Got graphics ready for the release.

![MC2-noVNC130](https://user-images.githubusercontent.com/1319013/143939950-cd36416d-569a-46c2-92c6-15beb7736537.png)

**Comment by Ylianst** [comment]
MeshCentral v0.9.55 is out with latest noVNC included. Let me know if it works and close this issue if it does. Thanks.

**Comment by AntonAndreevichMoroz** [comment]
Unfortunately, this fix was not included in release 1.3.0. But only to the master branch. I will follow, as soon as there is a release that includes support for ZRLE I will notify.

P.S. I just thought you were using not only the release versions. Hastened.

**Comment by si458** [comment]
@AntonAndreevichMoroz i have created a PR to update to 1.4.0 of noVNC, please can you try it? #5220
